components/data-provider/src/structs/dates.rs

```diff
@@ -0,0 +1,131 @@
+// Date
+#[cfg(feature = "invariant")]
+use crate::prelude::*;
+
+/// Gets a locale-invariant default struct given a data key in this module's category.
+#[cfg(feature = "invariant")]
+pub(crate) fn get_invariant(data_key: &DataKey) -> Option<DataResponse<'static>> {
+    use crate::invariant::make_inv_response;
+    if data_key.category != DataCategory::Dates {
+        return None;
+    }
+    match (data_key.sub_category.as_str(), data_key.version) {
+        ("gregorian", 1) => make_inv_response::<gregorian::DatesV1>(),
+        _ => None,
+    }
+}
+
+pub mod gregorian {
+    use serde::{Deserialize, Serialize};
+    use std::borrow::Cow;
+    #[derive(Debug, PartialEq, Clone, Deserialize, Serialize, Default)]
+    pub struct DatesV1 {
```

Comment:
Why did you make all the fields here optional?

Reply:
To make it smaller when we serialize data files which have only some of those fields.

Reply:
If you have separate keys, you wouldn't need the options, because you request data in smaller batches.

Reply:
I don't mean runtime - I mean on disk JSON files. `Option` makes it possible to serialize it as:

```json
{symbols: ...}
```

or:

```json
{patterns: ...}
```

instead of:

```json
{symbols: null, patterns: ...}
```

It may seem like not a lot, but it accrues, so I like to use `Option` for any value that can be missing, which is basically almost every field (since CLDR is a growing database and locales may not have fields).
I can see an argument that the foundational data (there's some subset of CLDR that is mandatory for a locale to be included) could not use options.

I also assume that you may mean here that if we split it into separate files then a file presence is an indicator serving the same goal and I agree with that. But we won't split every subkey into a separate file, so I'm just using Options here for that.
I suggested we start by landing all dates as a single key and discuss how to split it as we measure perf/size impact.

Reply:
See my comment in the main thread.  My concern isn't about `#[serde(skip_serializing_if = "Option::is_none")]`; my concern is about the notion that these fields are nullable in the first place.

Reply:
Done